packages/quantic/force-app/main/default/lwc/quanticNumericFacet/__tests__/quanticNumericFacet.test.js

```diff
@@ -229,11 +407,442 @@ describe('c-quantic-numeric-facet', () => {
         expect(facetContent).toBeNull();
       });
     });
+  });
 
-    afterAll(() => {
-      functionsMocks.buildNumericFacet.mockReturnValue(
-        numericFacetControllerMock
+  describe('when the component is loading', () => {
+    beforeEach(() => {
+      searchStatusState = {
+        ...initialFacetState,
+        isLoading: true,
+        hasError: false,
+        firstSearchExecuted: false,
+      };
+    });
+
+    it('should display the placeholder component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const placeholder = element.shadowRoot.querySelector(
+        selectors.placeholder
+      );
+      expect(placeholder).not.toBeNull();
+    });
+
+    describe('when the interface has an error', () => {
```

Comment:
I think this should/could be a top level describe no? When the interface has an error doesn't only concern the placeholder right?

Reply:
Writing the tests based on the code is also not something I would recommend. For example it raises questions such as "what happens to the tests if the code structure changes?"

The goal of those tests is to make us confident that the component still works even if the code of the component changes. So there's no reason to structure the tests according to the inner structure of the component's code.

The important part of this test is that we do not display the placeholder when the interface has an error, so IMO we should simply replace this test, with adding a check to a test in the component initialization maybe?

It can stay there as well, but I want to make sure that the reason is not because we want to respect the inner logic of the component itself.